### PR TITLE
Setup test suite

### PR DESCRIPTION
**Pull request by jhrcek**
This PR is just for inspiration on how to setup a test suite in a more standard way.

It's based on hspec which has nice [tutorial](https://hspec.github.io/getting-started.html).

The way to use this is to run `stack test --file-watch` which will recompile everything and rerun all tests whenever you save a file.

### CODE REVIEW

File: src/Week1/HW01Tests.hs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 module HW01Tests where
 
-import HW01
+import Week1.HW01
```
Comment (jhrcek) [inline src/Week1/HW01Tests.hs:7]: The module names should reflect the directory path starting from what you have in `hs-source-dirs` field of the library section in your cabal file. With this name stack / cabal will be able to discover the file if you run `stack repl` (or most other stack commands for that matter) from the root of your project.

File: test/Spec.hs [hunk 2]
```diff
@@ -1,2 +1 @@
-main :: IO ()
-main = putStrLn "Test suite not yet implemented"
+{-# OPTIONS_GHC -F -pgmF hspec-discover #-}
```
Comment (jhrcek) [inline test/Spec.hs:1]: This is just a trick to save you some boilerplate. Thanks to this hspec discovers all the tests modules without you having to manually create "root test file". Described here: https://hspec.github.io/hspec-discover.html

File: test/Week1Spec.hs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import Week1.HW01 
+import Test.Hspec
+
+spec :: Spec
+spec =
+    describe "Week 1" $ do
+        describe "lastDigit" $ do
+            it "should return last digit for a number" $
+                lastDigit 15 `shouldBe` 5
```
Comment (jhrcek) [inline test/Week1Spec.hs:11]: Using [shouldBe](https://hackage.haskell.org/package/hspec-expectations-0.8.2/docs/Test-Hspec-Expectations.html#v:shouldBe) and friends will save you some from having to create your own test framework.

**Comment by dzhou343** [comment]
Thank you for this! I will merge this in and incorporate this for all the homework.

**Pull request merged**